tools/nnyeah/nnyeah/AssemblyComparator/ComparingVisitor.cs

```diff
@@ -0,0 +1,144 @@
+			VisitMembers (reworker, earlier.Fields, later.Fields);
+			VisitMembers (reworker, earlier.Events, later.Events);
+			VisitMembers (reworker, earlier.Properties, later.Properties);
+		}
+
+		void VisitMembers<T> (TypeReworker reworker,
+			List<TypeElement<T>> earlier, List<TypeElement<T>> later) where T : IMemberDefinition
+		{
+			foreach (var earlierElem in earlier) {
+				VisitMember (reworker, earlierElem, later);
+			}
+		}
+
+		void VisitMember<T> (TypeReworker reworker, TypeElement<T> elem,
+			List<TypeElement<T>> later) where T : IMemberDefinition
+		{
+			foreach (var late in later) {
+				if (elem.Signature == late.Signature) {
+					FireElementFound (elem, late);
+					return;
+				}
+			}
+			var remappedSig = RemappedSignature (reworker, elem.Element);
+			if (remappedSig == elem.Signature) {
+				FireElementNotFound (elem);
+				return;
+			}
+			foreach (var late in later) {
+				if (remappedSig == late.Signature) {
+					FireElementFound (elem, late);
+					return;
+				}
+			}
+			FireElementNotFound (elem);
+		}
+
+		void FireElementNotFound<T> (TypeElement<T> earlier) where T : IMemberDefinition
+		{
+			if (earlier is TypeElement<FieldDefinition> field) {
+				FieldNotFound.Invoke (this, new FieldNotFoundEventArgs (field.Signature));
+			} else if (earlier is TypeElement<MethodDefinition> method) {
+				MethodNotFound.Invoke (this, new MethodNotFoundEventArgs (method.Signature));
+			} else if (earlier is TypeElement<EventDefinition> @event) {
+				EventNotFound.Invoke (this, new EventNotFoundEventArgs (@event.Signature));
+			} else if (earlier is TypeElement<PropertyDefinition> property) {
+				PropertyNotFound.Invoke (this, new PropertyNotFoundEventArgs (property.Signature));
+			}
+		}
+
+		void FireElementFound<T> (TypeElement<T> earlier, TypeElement<T> later) where T : IMemberDefinition
+		{
+			if (earlier is TypeElement<FieldDefinition> field) {
+				FieldFound.Invoke (this, new FieldFoundEventArgs (field.Signature, later.Signature));
+			} else if (earlier is TypeElement<MethodDefinition> method) {
+				MethodFound.Invoke (this, new MethodFoundEventArgs (method.Signature, later.Signature));
+			} else if (earlier is TypeElement<EventDefinition> @event) {
+				EventFound.Invoke (this, new EventFoundEventArgs (@event.Signature, later.Signature));
+			} else if (earlier is TypeElement<PropertyDefinition> property) {
+				PropertyFound.Invoke (this, new PropertyFoundEventArgs (earlier.Signature, later.Signature));
+			}
+		}
+
+		static string RemappedSignature<T> (TypeReworker reworker, T elem) where T : IMemberDefinition
+		{
+			switch (elem) {
+			case MethodDefinition method:
+				return reworker.ReworkMethod (method).ToString ();
+			case FieldDefinition field:
+				return reworker.ReworkField (field).ToString ();
+			case EventDefinition @event:
+				return reworker.ReworkEvent (@event).ToString ();
+			case PropertyDefinition property:
+				return reworker.ReworkProperty (property).ToString ();
```

Comment:
If we were going to leverage generics here, we could have one Rework with overloads for each type. If we're going to do type checking, we can get rid of the generics, as noted above.